github-runner-manager/src/github_runner_manager/manager/runner_manager.py

```diff
@@ -529,13 +547,60 @@ def wait_for_runner_online(
         for wait_time in RUNNER_CREATION_WAITING_TIMES:
             time.sleep(wait_time)
             try:
-                runner_health = platform_provider.get_runner_health(
-                    metadata=metadata, instance_id=instance_id
-                )
-            except PlatformApiError as exc:
-                logger.error("Error getting the runner health: %s", exc)
+                runner_health = platform_provider.get_runner_health(runner_identity)
+            except PlatformApiError:
+                logger.exception("Error getting the runner health: %s", runner_identity)
                 continue
             if runner_health.online or runner_health.deletable:
+                logger.info("Runner %s online", runner_identity)
                 break
+            logger.info("Runner %s not yet online", runner_identity)
         else:
-            raise RunnerError(f"Runner {instance_id} did not get online")
+            raise RunnerError(f"Runner {runner_identity} did not get online")
+
+
+def _filter_runner_to_delete(
+    cloud_runner: CloudRunnerInstance,
+    health: PlatformRunnerHealth | None,
+    *,
+    clean_idle: bool = False,
+    force_delete: bool = False,
+) -> bool:
+    """Filter runners to delete based on the input arguments.
+
+    This is the main function to filter what runners to delete. Runners that are deletable
+    in the health information from platform should be deleted. For the other cases, the
+    filtering will depend on the input arguments.
+
+    Args:
+        cloud_runner: Cloud runner.
+        health: Platform runner or None if health information is not available.
+        clean_idle: Remove runners that are idle.
+        force_delete: Delete the runner in all conditions.
+
+    Returns:
+        True if the runner should be deleted
+    """
+    if force_delete:
+        return True
+
+    # Do not delete runners without health information.
+    if health is None:
+        logger.info("No health information for %s. deleting: %s", health, force_delete)
+        return False
```

Comment:
We are returning False independently of force_delete. I think the log message has to be changed.